### PR TITLE
Add cue_up_with_data command

### LINKED ISSUE #18: add cue up to data command
This command requires a timecode input:
https://github.com/hideakitai/Sony9PinRemote/blob/20227fa426bc66fc182333bfbb0e9156a91b86e5/Sony9PinRemote/Encoder.h#L272-L288

### PR DESCRIPTION
**Pull request by g-maxime**
Fixes #18

Please test

Signed-off-by: Maxime Gervais <gervais.maxime@gmail.com>

### CODE REVIEW

**Comment by dericed** [comment]
It works in some cases, but there's a base-10 / base-16 issue somewhere.

```
$ ./sony9pin.app/Contents/MacOS/sony9pin -v 8 c 01:00:09:09
Info: verbose mode.
Info: open device cu.usbserial-AB0LXAZY.
Info: open device OK.
Info: cue_up_with_data.

$ ./sony9pin.app/Contents/MacOS/sony9pin 8 2
TimeCode: 00:00:09;09 CF: 0 DF: 1

$ ./sony9pin.app/Contents/MacOS/sony9pin -v 8 c 01:00:09:15
Info: verbose mode.
Info: open device cu.usbserial-AB0LXAZY.
Info: open device OK.
Info: cue_up_with_data.

$ ./sony9pin.app/Contents/MacOS/sony9pin 8 2
TimeCode: 00:00:09;09 CF: 0 DF: 1

$ ./sony9pin.app/Contents/MacOS/sony9pin -v 8 c 01:00:09:16
Info: verbose mode.
Info: open device cu.usbserial-AB0LXAZY.
Info: open device OK.
Info: cue_up_with_data.

$ ./sony9pin.app/Contents/MacOS/sony9pin 8 2
TimeCode: 00:00:09;10 CF: 0 DF: 1

$ ./sony9pin.app/Contents/MacOS/sony9pin -v 8 c 01:00:09:17
Info: verbose mode.
Info: open device cu.usbserial-AB0LXAZY.
Info: open device OK.
Info: cue_up_with_data.

$ ./sony9pin.app/Contents/MacOS/sony9pin 8 2
TimeCode: 00:00:09;11 CF: 0 DF: 1
```

File: tools/sony9pin/sony9pin.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@ void commands(const char* const prefix = "") {
     << prefix << "p: play\n"
     << prefix << "r: rewind\n"
     << prefix << "s: stop\n"
+    << prefix << "c <timecode in HH:mm:ss:ff format>: cue_up_with_data"
```
Comment (dericed) [inline tools/sony9pin/sony9pin.cpp:37]: Please add a terminating line break

```
s: stop
c <timecode in HH:mm:ss:ff format>: cue_up_with_data0: status
1: type
```

**Comment by JeromeMartinez** [comment]
@dericed it is updated (thanks @g-maxime!), please review again.

**Comment by dericed** [comment]
Seems the same. Upstream issue?

**Comment by g-maxime** [comment]
> Seems the same. Upstream issue?

No issue with the input conversion...

Is there a way to read the timecode outside sony9pin?  (A screen on the deck ?) to see if it's coherent with sony9pin output?

**Comment by dericed** [comment]
Hi @g-maxime, yes there is a screen on the deck and it matches this call:
```
$ ./sony9pin.app/Contents/MacOS/sony9pin 8 2
TimeCode: 00:00:09;10 CF: 0 DF: 1
```

**Comment by g-maxime** [comment]
> Hi @g-maxime, yes there is a screen on the deck and it matches this call:
> 
> ```
> $ ./sony9pin.app/Contents/MacOS/sony9pin 8 2
> TimeCode: 00:00:09;10 CF: 0 DF: 1
> ```

So the cue command don't sets the tape at the right timecode?

I can't figure out from the doc if the timecode needs to be bcd-encoded for the function https://github.com/hideakitai/Sony9PinRemote/blob/20227fa426bc66fc182333bfbb0e9156a91b86e5/Sony9PinRemote/Encoder.h#L272-L288

**Comment by dericed** [comment]
oooo, looks better

**Comment by dericed** [comment]
This works great and now I can get the frame I want if it exists.

**Pull request merged**